packages/markdown/remark/src/remark-collect-images.ts

```diff
@@ -1,42 +1,45 @@
-import type { Image, ImageReference } from 'mdast';
+import type { Root } from 'mdast';
 import { definitions } from 'mdast-util-definitions';
 import { visit } from 'unist-util-visit';
 import type { VFile } from 'vfile';
 
-export function remarkCollectImages() {
-	return function (tree: any, vfile: VFile) {
+interface Opts {
+	allowedRemoteDomains?: string[];
+}
+
+export function remarkCollectImages(opts?: Opts) {
+	const allowedRemoteDomains = new Set(opts?.allowedRemoteDomains ?? []);
+
+	return function (tree: Root, vfile: VFile) {
 		if (typeof vfile?.path !== 'string') return;
 
 		const definition = definitions(tree);
-		const imagePaths = new Set<string>();
-		visit(tree, ['image', 'imageReference'], (node: Image | ImageReference) => {
+		const localImagePaths = new Set<string>();
+		const remoteImagePaths = new Set<string>();
+		visit(tree, (node) => {
+			let url: string | undefined;
 			if (node.type === 'image') {
-				if (shouldOptimizeImage(node.url)) imagePaths.add(decodeURI(node.url));
-			}
-			if (node.type === 'imageReference') {
+				url = decodeURI(node.url);
+			} else if (node.type === 'imageReference') {
 				const imageDefinition = definition(node.identifier);
 				if (imageDefinition) {
-					if (shouldOptimizeImage(imageDefinition.url))
-						imagePaths.add(decodeURI(imageDefinition.url));
+					url = decodeURI(imageDefinition.url);
 				}
 			}
+
+			if (!url) return;
+
+			try {
+				const domain = new URL(url).host;
+				if (allowedRemoteDomains.has(domain)) remoteImagePaths.add(url);
+			} catch {
+				// Not a valid remote URL. Check if it's a local image. If it's an absolute path, then it's not.
+				if (!url.startsWith('/')) localImagePaths.add(url);
```

Comment:
A couple of comments:
- nit: can you stick to our internal style and use brackets after if statements?
- use `URL.canParse` instead of relying on a try/catch